src/main/kotlin/org/opensearch/replication/action/resume/TransportResumeIndexReplicationAction.kt

```diff
@@ -97,7 +99,16 @@ class TransportResumeIndexReplicationAction @Inject constructor(transportService
                     remoteClient.admin().indices()::getSettings,
                     injectSecurityContext = true
                 )(getSettingsRequest)
-                ValidationUtil.validateAnalyzerSettings(environment, settingsResponse.indexToSettings.get(params.leaderIndex.name), replMetdata.settings)
+
+                val leaderSettings = settingsResponse.indexToSettings.get(params.leaderIndex.name) ?: throw IndexNotFoundException(params.leaderIndex.name)
+
+                // k-NN Setting is a static setting. In case the setting is changed at the leader index before resume,
+                // block the resume.
+                if(leaderSettings.getAsBoolean(KNN_INDEX_SETTING, false)) {
+                    throw IllegalStateException("Index setting (index.knn) changed. Cannot resume k-NN index - ${params.leaderIndex.name}")
+                }
+
+                ValidationUtil.validateAnalyzerSettings(environment, leaderSettings, replMetdata.settings)
```

Comment:
Minor messaging feedback - "Cannot resume replication for k-NN enabled index {index}." ?

Reply:
Done